dune/grid/CpGrid.hpp

```diff
@@ -564,6 +571,27 @@ namespace Dune
             return 0;
         }
 
+        /// \brief returns the number of boundary segments within the macro grid
+        unsigned int numBoundarySegments() const
```

Comment:
It's fine with me. BTW is the parallel case handled correctly? Not sure what is expected of the grid in that case, and it looks to me like the implementation here may count faces that are interior to the global grid, and on the boundary of the local grid.